src/cursor.cpp

```diff
@@ -0,0 +1,46 @@
+#include "util.h"
+#include "voxel.cpp"
+
+class Cursor {
+  private:
+    tuple3i position;
+    tuple3i color;
+    int shape;
+
+  public:
+    void setShape (int flag) {
+      shape = flag;
+    }
+
+    tuple3i getPosition () {
+      return position;
+    }
+
+    tuple3i getColor () {
+      return color;
+    }
+
+    int getShape () {
+      return shape;
+    }
+
+    void setColor (tuple3i newColor) {
+      get<0>(color) = get<0>(newColor);
```

Comment:
for setting, the `=` operator should be enough according to http://www.cplusplus.com/reference/tuple/tuple/operator=/; it acts as a copy assignment. Getting and setting each member of the tuple individually is probably unnecessary. (I'm not sure though, since I haven't tested it.)